builds/e2e/upgrade-scenarios.yaml

```diff
@@ -0,0 +1,229 @@
+trigger: none
+pr: none
+
+resources:
+  pipelines:
+  - pipeline: images
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'main'
+    trigger:
+      branches:
+      - main
+      # TODO: Uncomment and remove "main" when release/1.3 is created
+      #- release/1.3
+  - pipeline: packages
+    source: 'Azure-IoT-Edge-Core Edgelet Packages'
+    branch: 'main'
+    trigger:
+      branches:
+      - main
+      # TODO: Uncomment and remove "main" when release/1.3 is created
+      #- release/1.3
+  - pipeline: images_release_1_1
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'release/1.1'
+  - pipeline: images_release_1_2
+    source: 'Azure-IoT-Edge-Core Build Images'
+    branch: 'release/1.2'
+
+jobs:
+################################################################################
+  - job: partial_upgrade_ubuntu_1804_msmoby
+################################################################################
+    displayName: Partial Upgrade
+
+    pool:
+      name: $(pool.linux.name)
+      demands:
+      - ImageOverride -equals agent-aziotedge-ubuntu-18.04-msmoby
+
+    variables:
+      os: linux
+      arch: amd64
+      artifactName: iotedged-ubuntu18.04-amd64
+      identityServiceArtifactName: packages_ubuntu-18.04_amd64
+      identityServicePackageFilter: aziot-identity-service_*_amd64.deb
+      minimal: true
+      verbose: false
+
+    strategy:
+      matrix:
+        # Scenarios 1 to 5 test older versions of the bootstrap Edge Agent 
+        # module with the 1.3 Edge Daemon and Runtime Modules. These represent
+        # scenarios where the Edge Daemon and Runtime Modules in the deployment 
+        # are upgraded, but the bootstrap agent in config.toml/config.yaml is not.
+        1.1.0_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.1.0"
+          runtimeModulesVersion: "release/1.3"
+        1.1.8_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.1.8"
+          runtimeModulesVersion: "release/1.3"
+        1.1.13_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.1.13"
+          runtimeModulesVersion: "release/1.3"
+        1.2.0_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.0"
+          runtimeModulesVersion: "release/1.3"
+        1.2.9_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.9"
```

Comment:
Now that 1.2.10 is out. Do we want to be using that? 
The first and the last version from the branch. haha.

Reply:
Agreed. Done.